test/unit/dashboard_test.rb

```diff
@@ -0,0 +1,36 @@
+require 'test_plugin_helper'
+
+class DashboardTest < ActiveSupport::TestCase
+  let(:report1) { FactoryBot.create(:host_report, :ansible_format) }
+  let(:report2) { FactoryBot.create(:host_report, :puppet_format) }
+  let(:report3) { FactoryBot.create(:host_report, :puppet_format) }
+
+  let(:host1) { report1.host }
+  let(:host2) { report2.host }
+  let(:host3) { report3.host }
+  let(:hosts) { [host1, host2, host3] }
+
+  test 'check number of host reports with changes' do
+    report1.update(:change => 1)
+    changed = report1.change + report2.change + report3.change
+    assert_equal HostReport.report_changes.count, changed
+  end
+
+  test 'check number of host reports with no changes' do
+    report2.update(:nochange => 1)
+    nochange = report1.nochange + report2.nochange + report3.nochange
+    assert_equal HostReport.report_nochanges.count, nochange
+  end
+
+  test 'check number of host reports with failures' do
+    report3.update(:failure => 1)
+    failure = report1.failure + report2.failure + report3.failure
+    assert_equal HostReport.report_failures.count, failure
+  end
+
+  test 'check Hosts with disabled alerts ' do
+    host1.update(:enabled => false)
```

Comment:
I would suggest you update the factory bot with some traits that can be shared across tests. We will likely get many more tests in the future.